src/sync/probspecs.nim

```diff
@@ -56,27 +57,33 @@ proc exercisesWithCanonicalData(repo: ProbSpecsRepo): seq[ProbSpecsRepoExercise]
 func slug(repoExercise: ProbSpecsRepoExercise): string =
   extractFilename(repoExercise.dir)
 
-func initProbSpecsTestCase(node: JsonNode): ProbSpecsTestCase =
-  result.json = node
+func initProbSpecsTestCase(node: JsonNode, parents: seq[JsonNode]): ProbSpecsTestCase =
+  ProbSpecsTestCase(json: node, parents: parents)
 
 proc uuid*(testCase: ProbSpecsTestCase): string =
   testCase.json["uuid"].getStr()
 
 proc description*(testCase: ProbSpecsTestCase): string =
-  testCase.json["description"].getStr()
+  for parent in testCase.parents:
+    if parent.contains("description"):
+      result.add(parent["description"].getStr() & " → ")
+
+  result.add(testCase.json["description"].getStr())
 
 func isReimplementation*(testCase: ProbSpecsTestCase): bool =
   testCase.json.hasKey("reimplements")
 
 proc reimplements*(testCase: ProbSpecsTestCase): string =
   testCase.json["reimplements"].getStr()
 
-proc initProbSpecsTestCases(node: JsonNode): seq[ProbSpecsTestCase] =
+proc initProbSpecsTestCases(node: JsonNode, parents: seq[JsonNode] = @[]): seq[ProbSpecsTestCase] =
   if node.hasKey("uuid"):
-    result.add(initProbSpecsTestCase(node))
+    result.add(initProbSpecsTestCase(node, parents))
   elif node.hasKey("cases"):
+    var newParents = parents
+    newParents.add(node)
```

Comment:
I don't know if this is the nicest way to do this, or should I make the argument mutable?